docs/csharp/tutorials/working-with-linq.md

```diff
@@ -80,64 +77,97 @@ static IEnumerable<string> Ranks()
 
-This method declaration also follows a standard idiom where the input and output types are `IEnumerable<T>`. That practice enables LINQ methods to be chained together to perform more complex queries.
+Let's give our extension methods a home: add a new *static* class file to your program called `Extensions.cs`, and then start building out our first extension method: 
 
 ```csharp
+// Extensions.cs
+using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace LinqFaroShuffle
 {
     public static class Extensions
     {
-        public static IEnumerable<T> InterleaveSequenceWith<T>
-            (this IEnumerable<T> first, IEnumerable<T> second)
+        public static IEnumerable<T> InterleaveSequenceWith<T>(this IEnumerable<T> first, IEnumerable<T> second)
         {
-            // implementation coming.
+            // Our implementation will go here soon enough
         }
     }
 }
 ```
 
-You will be enumerating both sequences at once, interleaving the elements, and creating one object.  Writing a LINQ method that works with two sequences requires that you understand how `IEnumerable` works.
+Look at the method signature for a moment, specifically the parameters:
 
-The `IEnumerable` interface has one method: `GetEnumerator()`. The object returned by `GetEnumerator()` has a method to move to the next element, and a property that retrieves the current element in the sequence. You will use those two members to enumerate the collection and return the elements. This Interleave method will be an iterator method, so instead of building a collection and returning the collection, you'll use the `yield return` syntax shown above. 
+```csharp
+public static IEnumerable<T> InterleaveSequenceWith<T>
+    (this IEnumerable<T> first, IEnumerable<T> second)
+```
+
+You can see the addition of the `this` modifier on the first argument to the method. That means you call the method as though it were a member method of the type of the first argument. This method declaration also follows a standard idiom where the input and output types are `IEnumerable<T>`. That practice enables LINQ methods to be chained together to perform more complex queries.
+
+Naturally, since we split the deck into halves, we'll need to join those halves together. In code, this means we'll be enumerating both of the sequences we acquired through `Take()` and `Skip()` at once, *interleaving* the elements, and creating one object: our now-shuffled deck of cards. Writing a LINQ method that works with two sequences requires that you understand how `IEnumerable` works.
```

Comment:
At the end of the paragraph: `IEnumerable` -> `IEnumerable<T>`, unless you really mean the non-generic `IEnumerable` interface.

Reply:
Yeah, that's a good point. I believe I left it as-is because the original doc also had it as a plain `IEnumerable`. I'll make sure that it's the generic interface here and see if I forgot to include it elsewhere.